tests/test_app.py

```diff
@@ -8,6 +9,20 @@ def test_init():
     assert app
 
 
+def test_run():
+    app = Application()
+    app.run_internally = mock.MagicMock(return_value=True)
+    assert app.run()
+    assert app.run_internally.called
+
+
+def test_run_with_exception():
```

Comment:
And in this test, just check the return value. I suggest 3 following tests:

 * `test_run_calls_run_internally` (use mock)
 * `test_run_returns_true_on_success` (use lambda)
 * `test_run_returns_false_on_exception` (use lambda)